src/devtools/client/debugger/src/components/Editor/ColumnBreakpoints.tsx

```diff
@@ -26,33 +29,58 @@ type PropsFromRedux = ConnectedProps<typeof connector>;
 
 type $FixTypeLater = any;
 interface CBProps {
-  editor: $FixTypeLater;
+  editor: SourceEditor;
 }
 
-class ColumnBreakpoints extends Component<PropsFromRedux & CBProps> {
-  render() {
-    const { cx, editor, columnBreakpoints, selectedSource } = this.props;
-
-    if (!selectedSource || columnBreakpoints.length === 0) {
-      return null;
-    }
-
-    let breakpoints;
-    // TODO Is this a safe thing to do while rendering?
-    editor.codeMirror.operation(() => {
-      breakpoints = columnBreakpoints.map((breakpoint, i) => (
-        <ColumnBreakpoint
-          cx={cx}
-          key={getLocationKey(breakpoint.location)}
-          columnBreakpoint={breakpoint}
-          editor={editor}
-          source={selectedSource}
-          insertAt={i}
-        />
-      ));
-    });
-    return <div>{breakpoints}</div>;
+function ColumnBreakpoints2({ editor }: CBProps) {
+  const cx = useAppSelector(getContext);
+  const selectedSource = useAppSelector(getSelectedSource);
+  const columnBreakpoints = useAppSelector(getVisibleColumnBreakpoints);
+  const { value: hitCountsMode } = useStringPref("hitCounts");
+
+  useEffect(() => {
+    const updateWidth = () => {
+      const editorElement = editor.editor.getWrapperElement();
+      const gutter = editor.editor.getGutterElement();
+
+      const gutterWidth = gutter.getBoundingClientRect().width;
+      const scrollWidth = editorElement.getBoundingClientRect().width;
+
+      // Magic formula to determine scrollbar width: https://stackoverflow.com/a/60008044/62937
+      let scrollbarWidth = window.innerWidth - document.body.clientWidth;
+
+      // Fill the _visible_ width of the editor, minus scrollbar, minus a small bit of padding
+      const newWidth = scrollWidth - gutterWidth - scrollbarWidth - 10;
+
+      let root = document.documentElement;
+      root.style.setProperty("--print-statement-max-width", newWidth + "px");
+    };
+
+    editor.editor.on("refresh", updateWidth);
+
+    updateWidth();
+
+    return () => {
+      editor.editor.off("refresh", updateWidth);
+    };
+  }, [editor, hitCountsMode]);
```

Comment:
This looks fine. We could also maybe do it via uSES too but no preference which we use really.
https://github.com/replayio/devtools/pull/7837#discussion_r983410954